src/hooks/useViewport/useViewport.ts

```diff
@@ -0,0 +1,38 @@
+import { useLayoutEffect, useState, useContext } from 'react';
+import throttle from 'lodash.throttle';
+import { ViewportContext } from './ViewportProvider';
+
+export interface IUseViewportOptions {
+  /**
+   * The timeout supplied to `lodash.throttle` in case the viewport dimensions are read directly.
+   */
+  timeout?: number;
+}
+
+function readViewport() {
+  const width = Math.max(document.documentElement.clientWidth, window.innerWidth || 0);
+  const height = Math.max(document.documentElement.clientHeight, window.innerHeight || 0);
+
+  return {
+    height,
+    width,
+  };
+}
+
+export function useViewport({ timeout = 300 }: IUseViewportOptions = {}) {
+  const contextSize = useContext(ViewportContext);
+  if (contextSize !== undefined) return contextSize;
+
+  const [size, setSize] = useState(contextSize || readViewport());
```

Comment:
If you are doing ` if (contextSize !== undefined) return contextSize;`

when this condition is going to happen? `useState(contextSize || readViewport())`

Reply:
ahhh good, I think it should be here just: `useState(readViewport)` 👌🏻

Reply:
```suggestion
  const [size, setSize] = useState(readViewport());
```